site/src/components/Workspace/Workspace.tsx

```diff
@@ -65,6 +67,8 @@ export const Workspace: FC<React.PropsWithChildren<WorkspaceProps>> = ({
 }) => {
   const styles = useStyles()
   const navigate = useNavigate()
+  const xServices = useContext(XServiceContext)
+  const [buildInfoState] = useActor(xServices.buildInfoXService)
```

Comment:
Saw your comment about Storybook and XState. That's why we have XState only in the pages right now. There must be a way to set up Storybook to get around it, but we haven't put the time in yet (just adding the XServiceProvider to the Storybook decorator doesn't get us all the way there), and I generally like that it makes us separate "connected" components from visual ones. The problem is in the workspace page, where there is so much that we end up with a deep component tree. Our solution so far has been to send things from XState down from the WorkspacePage as props but make objects of some bundles of props to make it less messy.

Reply:
Gotcha, that makes sense. I'll go ahead with that approach for now!

Reply:
Oh wait, I was doing an import wrong - I think just adding the provider does work! If you want to try, the place to add it is .storybook/preview.js.